BaseClasses.py

```diff
@@ -693,9 +693,10 @@ def can_reach(self,
                 spot = self.multiworld.get_region(spot, player)
         return spot.can_reach(self)
 
-    def sweep_for_events(self, key_only: bool = False, locations: Optional[Iterable[Location]] = None) -> None:
+    def sweep_for_events(self, key_only: bool = False, locations: Optional[Iterable[Location]] = None,
+                         player: Optional[int] = None) -> None:
```

Comment:
this seems like a doubling of API responsibilities.  Surely the same result can be achieved by just passing in a single player's locations?

Reply:
that's what it does already. i suppose i could remove that arg and have the method do a check that all locations belong to only a single player, but that would be slower so i'm not sure if it's worth it?

Reply:
reread it and understand what you meant now